app/models/proxy.rb

```diff
@@ -529,49 +530,71 @@ def configuration_is_object
   class PoliciesConfig
     include ActiveModel::Validations
 
-    delegate :each, to: :policies_config
+    delegate :each, :find, :select, :map, :detect, :reject, to: :policies_config
     attr_reader :policies_config
+    protected :policies_config
 
     validate :policies_configs_are_correct
 
     def initialize(policies_config)
-      @policies_config = policies_config.map { |attrs| PolicyConfig.new(attrs) }
+      parsed = policies_config.presence || []
+      parsed = parsed.is_a?(Array) ? parsed.as_json : Array(JSON.parse(parsed))
+
+      @policies_config = policies_normalize(parsed)
+    rescue JSON::ParserError
```

Comment:
I've just tried to preserve original error messages and behavior.

wrt parsing I tried to strictly follow what we had in the past, just had to remove extra methods because of codeclimate errors, my desire was to keep as before for easier review and fewer changes. In the past you could get such invalid JSON and there are tests for invalid JSON input. We can of course change that on your request. I made existing tests pass. e.g. when invalid JSON is read from database or API call.

wrt duplicates, I can't spot where in the past we checked for duplicates. So it will be a new requirement to avoid duplicates. I think it's worth doing as a separate PR.

Please let me know how to proceed (and will write YARD once I know).